src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -229,6 +230,11 @@ internal CookieContainer CookieContainer
             set
             {
                 CheckDisposedOrStarted();
+                if (value == null)
+                {
+                    throw new ArgumentNullException("CookieContainer", SR.net_http_invalid_cookiecontainer);
```

Comment:
Created a bug - #3205.  Actually I also introduced a new error message but am not using it. The exception that will be thrown is same as what the winhttp throws. The message could be a little confusing: 
 
    System.InvalidOperationException: When using CookieUsePolicy.UseSpecifiedCookieContainer, the CookieContainer property must not be null